cmd/state-svc/internal/resolver/analytics/client.go

```diff
@@ -0,0 +1,75 @@
+package analytics
+
+import (
+	"runtime/debug"
+	"sync"
+
+	"github.com/ActiveState/cli/internal/analytics/deferred"
+	"github.com/ActiveState/cli/internal/analytics/event"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+)
+
+// Client is an AnalyticsDispatcher that is supposed to be use by the `state-svc` service only. It sends the events directly to the analytics event loop started in the resolver (once configured)
+type Client struct {
+	auth           *authentication.Auth
+	events         chan<- event.EventData
+	eventWaitGroup *sync.WaitGroup
+}
+
+func NewClient() *Client {
+	return &Client{auth: authentication.LegacyGet(), eventWaitGroup: &sync.WaitGroup{}}
+}
+
+func (c *Client) Event(category string, action string) {
+	c.EventWithLabel(category, action, "")
+}
+
+func (c *Client) EventWithLabel(category string, action, label string) {
+	err := c.sendEvent(category, action, label)
+	if err == nil {
+		return
+	}
+	logging.Error("Error during analytics.sendEvent: %v", errs.Join(err, ":"))
+}
+
+// Configure ties this client to the events loop running as part of the Resolver, un-configured clients defer events to the hard-drive
+func (c *Client) Configure(r *Resolver) {
+	c.events = r.events
+}
+
+func (c *Client) Wait() {
+	c.eventWaitGroup.Wait()
+}
+
+func (c *Client) sendEvent(category, action, label string) error {
+	userID := ""
+	if c.auth != nil && c.auth.UserID() != nil {
+		userID = string(*c.auth.UserID())
+	}
+
+	// For now analytics events triggered by the state-svc are NEVER bound to a project or an output-type
+	ev := event.New(category, action, &label, nil, nil, &userID)
+
+	// if events channel is not set yet, we will defer the events to the file system
+	if c.events == nil {
+		if err := deferred.DeferEvent(ev); err != nil {
+			return locale.WrapError(err, "err_analytics_defer", "Could not defer event")
+		}
+		return nil
+	}
```

Comment:
Please get rid of any deferring altogether. Deferring was a legacy workaround for not having the state-svc. If you still see use-cases for it please talk to me on Slack so we can come to a better solution.

Reply:
If we get rid of the deferring, everything will be much simpler...  I can get rid of the event loop too then.  I'll refactor it.